src/Components/Locations/LocationsContainer.js

```diff
@@ -0,0 +1,51 @@
+import React, {useState }from "react";
+import LocationCard from "./LocationCard";
+
+function LocationsContainer({ locations, setLocations }) {
+
+  // const [sortNames, setSortNames] = useState(false)
+  // const sortName = (e) => {
+  //   e.preventDefault()
+  //   setSortNames(!sortNames)
+  // }
+  // const [ key, setKey ] = useState("")
+
+  // const sortedByName = locations.sort((a,b) => (a.name < b.name ? 1 : -1))
+  // const sortedByClimate = locations.sort((a,b)=> (a.key < b.climate ? 1 : -1))
+  // const mappedByTerrain = locations.sort((a,b)=> (a.terrain < b.terrain ? 1 : -1))
+  // console.log(sortedByName)
+  
```

Comment:
1. make a new array from the state of locations; [...locations]
2.  create a search function that accepts a parameter - key sortStuff(key)
3.  .sort(a[key] < b[key] return 1 )